modules/economy.js

```diff
@@ -1,295 +1,255 @@
+                  channel.send(StringPlaceholder(action.message, processedInputs, { before: '#', after: '#' }) || 'Nothing')
+                  if (callback) callback()
+                } else {
+                  if (callback) callback(Error('ECO_ATI05'))
+                }
+              })
+            } else {
+              if (callback) callback(Error('ECO_ATI07'))
             }
-          })
-          if (propertiesString) {
-            callback(i18n(guild.locale, 'BUYPRODUCT_MISSING_PROPERTY', { PROPERTY: propertiesString }), false)
-          } else {
-            module.exports.addItemToUser(client, member, guild, item.productId, item.productMeta.singlebuy, (status) => {
-              if (status === 'error') {
-                callback(i18n(guild.locale, 'BUYPRODUCT_ALREADYOWN', { PRODUCT: item.productName }), false)
-              } else {
-                if (action && Object.prototype.hasOwnProperty.call(action, 'type')) {
-                  switch (action.type) {
-                    case 'sendMessage': {
-                      if (Object.prototype.hasOwnProperty.call(action, 'message')) {
-                        if (Object.prototype.hasOwnProperty.call(action, 'channel')) {
-                          action.message = StringPlaceholder(action.message, userInputs, { before: '#', after: '#' })
-                          guild.channels.fetch(action.channel).then(channel => {
-                            if (channel) {
-                              channel.send(action.message || 'Nothing', true)
-                              callback(null, true)
-                            } else {
-                              callback(i18n(guild.locale, 'BUYPRODUCT_SENDMESSAGE_ERROR'), false)
-                            }
-                          })
-                        } else {
-                          callback(i18n(guild.locale, 'BUYPRODUCT_SENDMESSAGE_ERROR'), false)
-                        }
-                      }
-                      break
-                    }
-                    case 'giveRole': {
-                      if (Object.prototype.hasOwnProperty.call(action, 'role')) {
-                        guild.roles.fetch(action.role).then(role => {
-                          if (role) {
-                            member.roles.add(role)
-                            callback(i18n(guild.locale, 'BUYPRODUCT_GIVEROLE', { ROLE: role }), true)
-                          } else {
-                            callback(i18n(guild.locale, 'BUYPRODUCT_GIVEROLE_ERROR'), false)
-                          }
-                        })
-                      } else {
-                        callback(i18n(guild.locale, 'BUYPRODUCT_GIVEROLE_ERROR'), false)
-                      }
-                    }
-                  }
+            break
+          }
+          case 'giveRole': {
+            if (Object.prototype.hasOwnProperty.call(action, 'role')) {
+              guild.roles.fetch(action.role).then(role => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/alexandrumuntenas/pingu/pullRequest?prid=8657799)
```suggestion
                        guild.roles.fetch(action.role).then((role) => {
```